File: frontend/lib/models/purchase_order_acceptance_request_dto.dart
class PurchaseOrderAcceptanceRequestDto {
  final List<PartAcceptanceDto> parts;
  final String supplierInn;
  final String? supplierName;
  final String? supplierContactPerson;
  final String? supplierPhone;
  final String? supplierEmail;
  final bool? supplierVerified;

  PurchaseOrderAcceptanceRequestDto({
    required this.parts,
    required this.supplierInn,
    this.supplierName,
    this.supplierContactPerson,
    this.supplierPhone,
    this.supplierEmail,
    this.supplierVerified,
  });

  Map<String, dynamic> toJson() => {
        'parts': parts.map((e) => e.toJson()).toList(),
        'supplierInn': supplierInn,
        'supplierName': supplierName,
        'supplierContactPerson': supplierContactPerson,
        'supplierPhone': supplierPhone,
        'supplierEmail': supplierEmail,
        'supplierVerified': supplierVerified,
      };
}

class PartAcceptanceDto {
  final int partId;
  final String status;
  final int acceptedQuantity;
  final String? comment;
  final String? storageLocation;
  final String? shelfCode;
  final String? cellCode;
  final String? placementNotes;

  PartAcceptanceDto({
    required this.partId,
    required this.status,
    required this.acceptedQuantity,
    this.comment,
    this.storageLocation,
    this.shelfCode,
    this.cellCode,
    this.placementNotes,
  });

  Map<String, dynamic> toJson() => {
        'partId': partId,
        'status': status,
        'acceptedQuantity': acceptedQuantity,
        'comment': comment,
        'storageLocation': storageLocation,
        'shelfCode': shelfCode,
        'cellCode': cellCode,
        'placementNotes': placementNotes,
      };
}

File: backend/src/main/java/ru/bowling/bowlingapp/DTO/PurchaseOrderAcceptanceRequestDTO.java
package ru.bowling.bowlingapp.DTO;

import jakarta.validation.Valid;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.bowling.bowlingapp.Entity.enums.PartStatus;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PurchaseOrderAcceptanceRequestDTO {
    @NotEmpty
    @Valid
    private List<PartAcceptanceDTO> parts;

    @NotBlank
    private String supplierInn;

    private String supplierName;

    private String supplierContactPerson;

    private String supplierPhone;

    private String supplierEmail;

    private Boolean supplierVerified;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class PartAcceptanceDTO {
        @NotNull
        private Long partId;

        @NotNull
        private PartStatus status;

        @Min(0)
        private Integer acceptedQuantity;

        private String comment;

        private String storageLocation;

        private String shelfCode;

        private String cellCode;

        private String placementNotes;
    }
}
